fastlane/lib/fastlane/helper/sh_helper.rb

```diff
@@ -68,5 +68,15 @@ def self.sh_control_output(command, print_command: true, print_command_output: t
       Encoding.default_external = previous_encoding.first
       Encoding.default_internal = previous_encoding.last
     end
+
+    def self.command_from_args(*args)
+      if args.count == 1
+        command = args.first
+        command = command.shelljoin if command.kind_of?(Array)
```

Comment:
Is this to account for nested arrays? i.e. someone calls `sh [["git", "commit", "-m", "my message"]]` this is what makes it work?

Reply:
@milch You're probably familiar with variadic Ruby methods, but for anyone else looking, here's how it works:

```Ruby
def sh(*command, log: true, error_callback: nil)
```

defines a variadic method that can take any number of arguments as well as named options at the end. The `command` arg is always received by the method as an Array, regardless how many args are passed. There is special handling for the case of one element in the array. This applies when you have:

```Ruby
sh "git commit -m #{message.shellescape}" # line 74
sh ["git", "commit", "-m", message] # line 75
```

Without the special handling for one element (if we called `args.shelljoin` for all input), things like `"git commit"` would be shell escaped to `"git\ commit"`, which would break

If the array has multiple elements, the array is shelljoined, so the case for `count != 1` handles this:

```Ruby
sh "git", "commit", "-m", message # line 77
```

The case with one element is important mainly for current Fastfiles and anything else that is building a shell-escaped string (or doesn't require a shell escape because no special chars, spaces in args, etc.). There can also be times when you have a string from somewhere that is a full command. Maybe the user types it into a config. file or something. Then you'd have to shellsplit the string in order to pass it into this method if it didn't handle a single string argument. 

The case where `command.count == 1 && command.first.kind_of?(Array)` handles `sh ["git", "commit", ...]`. Since support for this was only introduced yesterday, it's less critical, but there can be times where you build your command as an array and just want to pass the array. (Otherwise you'd have to pass `*array`.) And it can be nice to use delimiters to keep track of a lot of args:

```Ruby
sh ["git", "commit", "-m", message, path1, path2, path3], error_callback: ->(result) { ... }
```

There is one small thing I realized while writing this, which is when `command.count == 0`, you end up with a blank string. If you just write:

```Ruby
sh
```

It will attempt to execute `""`. This is probably not a major problem, but it would perhaps be best to raise `ArgumentError` if nothing is passed. I'll add a commit for that.

Reply:
Alright, thanks for the in-depth explanation. I wasn't sure wether the `*` operator would also wrap a single array argument in a second array, but from your explanation it seems like it does. I'll take another look once you've added the commit that handles the case when there are no arguments